### ISSUE TITLE
bug: gnome freezes in Uakari 6.6

### ISSUE THREAD

**Issue by aleeusgr** [aleeusgr/nix-things#79]
after nixos-rebuild switch gnome started to freeze.

```shell
$ dmesg
[   13.426960] nouveau 0000:2e:00.0: bus: MMIO read of 00000000 FAULT at 122124 [ PRIVRING ]
. . . 
[ 1112.286456]  entry_SYSCALL_64_after_hwframe+0x6e/0xd8
[ 1112.286457] RIP: 0033:0x7f5b9acac9af
[ 1112.286478] Code: 00 48 89 44 24 18 31 c0 48 8d 44 24 60 c7 04 24 10 00 00 00 48 89 44 24 08 48 8d 44 24 20 48 89 44 24 10 b8 10 00 00 00 0f 05 <89> c2 3d 00 f0 ff ff 77 18 48 8b 44 24 18 64 48 2b 04 25 28 00 00
[ 1112.286479] RSP: 002b:00007ffea6f76e80 EFLAGS: 00000246 ORIG_RAX: 0000000000000010
[ 1112.286480] RAX: ffffffffffffffda RBX: 0000000001ab07e0 RCX: 00007f5b9acac9af
[ 1112.286481] RDX: 00007ffea6f76f20 RSI: 00000000c0306480 RDI: 0000000000000009
[ 1112.286481] RBP: 00007ffea6f76f20 R08: 0000000000000004 R09: 0000000000000001
[ 1112.286482] R10: 0000000000000004 R11: 0000000000000246 R12: 00000000c0306480
[ 1112.286483] R13: 0000000000000009 R14: 00000000017e08a0 R15: 0000000001a9c708
[ 1112.286484]  </TASK>
[ 1112.286484] ---[ end trace 0000000000000000 ]---

```
and
```shell
$ journalctl -b
Apr 07 09:19:59 nixos kernel: [Firmware Bug]: TSC ADJUST differs within socket(s), fixing all errors
Apr 07 09:19:59 nixos kernel: RAS: Correctable Errors collector initialized.
Apr 07 09:19:59 nixos kernel: ACPI BIOS Error (bug): Could not resolve symbol [\CTDP], AE_NOT_FOUND (20230628/psargs-330)
Apr 07 09:19:59 nixos kernel: ACPI Error: Aborting method \_SB.IETM.IDSP due to previous error (AE_NOT_FOUND) (20230628/psparse-529)
Apr 07 09:19:59 nixos kernel: ACPI BIOS Error (bug): Could not resolve symbol [\_SB.PC00.LPCB.EC0.SEN1._CRT.S1CT], AE_NOT_FOUND (20230628/psargs-330)
Apr 07 09:19:59 nixos kernel: ACPI Error: Aborting method \_SB.PC00.LPCB.EC0.SEN1._CRT due to previous error (AE_NOT_FOUND) (20230628/psparse-529)
Apr 07 09:19:59 nixos kernel: ACPI BIOS Error (bug): Could not resolve symbol [\_SB.PC00.LPCB.EC0.SEN1._HOT.S1HT], AE_NOT_FOUND (20230628/psargs-330)
Apr 07 09:19:59 nixos kernel: ACPI Error: Aborting method \_SB.PC00.LPCB.EC0.SEN1._HOT due to previous error (AE_NOT_FOUND) (20230628/psparse-529)
Apr 07 09:19:59 nixos kernel: ACPI BIOS Error (bug): Could not resolve symbol [\_SB.PC00.LPCB.EC0.SEN1._PSV.S1PT], AE_NOT_FOUND (20230628/psargs-330)
Apr 07 09:19:59 nixos kernel: ACPI Error: Aborting method \_SB.PC00.LPCB.EC0.SEN1._PSV due to previous error (AE_NOT_FOUND) (20230628/psparse-529)
Apr 07 09:19:59 nixos kernel: ACPI BIOS Error (bug): Could not resolve symbol [\_SB.PC00.LPCB.EC0.SEN1._AC0.S1AT], AE_NOT_FOUND (20230628/psargs-330)
Apr 07 09:19:59 nixos kernel: ACPI Error: Aborting method \_SB.PC00.LPCB.EC0.SEN1._AC0 due to previous error (AE_NOT_FOUND) (20230628/psparse-529)
Apr 07 09:19:59 nixos kernel: ACPI BIOS Error (bug): Could not resolve symbol [\_SB.PC00.LPCB.EC0.SEN2._CRT.S2CT], AE_NOT_FOUND (20230628/psargs-330)
Apr 07 09:19:59 nixos kernel: ACPI Error: Aborting method \_SB.PC00.LPCB.EC0.SEN2._CRT due to previous error (AE_NOT_FOUND) (20230628/psparse-529)
Apr 07 09:19:59 nixos kernel: ACPI BIOS Error (bug): Could not resolve symbol [\_SB.PC00.LPCB.EC0.SEN2._HOT.S2HT], AE_NOT_FOUND (20230628/psargs-330)
Apr 07 09:19:59 nixos kernel: ACPI Error: Aborting method \_SB.PC00.LPCB.EC0.SEN2._HOT due to previous error (AE_NOT_FOUND) (20230628/psparse-529)
Apr 07 09:19:59 nixos kernel: ACPI BIOS Error (bug): Could not resolve symbol [\_SB.PC00.LPCB.EC0.SEN2._PSV.S2PT], AE_NOT_FOUND (20230628/psargs-330)
Apr 07 09:19:59 nixos kernel: ACPI Error: Aborting method \_SB.PC00.LPCB.EC0.SEN2._PSV due to previous error (AE_NOT_FOUND) (20230628/psparse-529)
Apr 07 09:19:59 nixos kernel: ACPI BIOS Error (bug): Could not resolve symbol [\_SB.PC00.LPCB.EC0.SEN2._AC0.S2AT], AE_NOT_FOUND (20230628/psargs-330)
Apr 07 09:19:59 nixos kernel: ACPI Error: Aborting method \_SB.PC00.LPCB.EC0.SEN2._AC0 due to previous error (AE_NOT_FOUND) (20230628/psparse-529)
Apr 07 09:19:59 nixos kernel: ACPI BIOS Error (bug): Could not resolve symbol [\_SB.PC00.LPCB.EC0.SEN3._CRT.S3CT], AE_NOT_FOUND (20230628/psargs-330)
Apr 07 09:19:59 nixos kernel: ACPI Error: Aborting method \_SB.PC00.LPCB.EC0.SEN3._CRT due to previous error (AE_NOT_FOUND) (20230628/psparse-529)
Apr 07 09:19:59 nixos kernel: ACPI BIOS Error (bug): Could not resolve symbol [\_SB.PC00.LPCB.EC0.SEN3._HOT.S3HT], AE_NOT_FOUND (20230628/psargs-330)
Apr 07 09:19:59 nixos kernel: ACPI Error: Aborting method \_SB.PC00.LPCB.EC0.SEN3._HOT due to previous error (AE_NOT_FOUND) (20230628/psparse-529)
Apr 07 09:19:59 nixos kernel: ACPI BIOS Error (bug): Could not resolve symbol [\_SB.PC00.LPCB.EC0.SEN3._PSV.S3PT], AE_NOT_FOUND (20230628/psargs-330)
Apr 07 09:19:59 nixos kernel: ACPI Error: Aborting method \_SB.PC00.LPCB.EC0.SEN3._PSV due to previous error (AE_NOT_FOUND) (20230628/psparse-529)
Apr 07 09:19:59 nixos kernel: ACPI BIOS Error (bug): Could not resolve symbol [\_SB.PC00.LPCB.EC0.SEN4._CRT.S4CT], AE_NOT_FOUND (20230628/psargs-330)
Apr 07 09:19:59 nixos kernel: ACPI Error: Aborting method \_SB.PC00.LPCB.EC0.SEN4._CRT due to previous error (AE_NOT_FOUND) (20230628/psparse-529)
Apr 07 09:19:59 nixos kernel: ACPI BIOS Error (bug): Could not resolve symbol [\_SB.PC00.LPCB.EC0.SEN4._HOT.S4HT], AE_NOT_FOUND (20230628/psargs-330)
Apr 07 09:19:59 nixos kernel: ACPI Error: Aborting method \_SB.PC00.LPCB.EC0.SEN4._HOT due to previous error (AE_NOT_FOUND) (20230628/psparse-529)
Apr 07 09:19:59 nixos kernel: ACPI BIOS Error (bug): Could not resolve symbol [\_SB.PC00.LPCB.EC0.SEN4._PSV.S4PT], AE_NOT_FOUND (20230628/psargs-330)
Apr 07 09:19:59 nixos kernel: ACPI Error: Aborting method \_SB.PC00.LPCB.EC0.SEN4._PSV due to previous error (AE_NOT_FOUND) (20230628/psparse-529)
Apr 07 09:19:59 nixos kernel: ACPI BIOS Error (bug): Could not resolve symbol [\_SB.PC00.LPCB.EC0.SEN4._AC0.S4AT], AE_NOT_FOUND (20230628/psargs-330)
Apr 07 09:19:59 nixos kernel: ACPI Error: Aborting method \_SB.PC00.LPCB.EC0.SEN4._AC0 due to previous error (AE_NOT_FOUND) (20230628/psparse-529)
Apr 07 09:20:01 nixos gnome-session[1301]: gnome-session-binary[1301]: WARNING: Could not check if unit gnome-session-wayland@gnome-login.target is active: Error calling StartServiceByName for org.freedesktop.systemd1: Process org.freedesktop.systemd1 exited with status 1
Apr 07 09:20:01 nixos gnome-session-binary[1301]: WARNING: Could not check if unit gnome-session-wayland@gnome-login.target is active: Error calling StartServiceByName for org.freedesktop.systemd1: Process org.freedesktop.systemd1 exited with status 1
Apr 07 09:20:01 nixos .gnome-shell-wr[1316]: Failed to make thread 'KMS thread' realtime scheduled: GDBus.Error:org.freedesktop.DBus.Error.NameHasNoOwner: Name "org.freedesktop.RealtimeKit1" does not exist
Apr 07 09:20:02 nixos .gnome-shell-wr[1316]: Error looking up permission: GDBus.Error:org.freedesktop.portal.Error.NotFound: No entry for geolocation
Apr 07 09:20:02 nixos .gsd-sharing-wr[1482]: Failed to StopUnit service: GDBus.Error:org.freedesktop.DBus.Error.Spawn.ChildExited: Process org.freedesktop.systemd1 exited with status 1
Apr 07 09:20:02 nixos .gsd-sharing-wr[1482]: Failed to StopUnit service: GDBus.Error:org.freedesktop.DBus.Error.Spawn.ChildExited: Process org.freedesktop.systemd1 exited with status 1
Apr 07 09:20:02 nixos org.gnome.Shell.desktop[1572]: Errors from xkbcomp are not fatal to the X server
Apr 07 09:20:02 nixos .geoclue-wrappe[1598]: Failed to open /etc/geoclue/conf.d/: Error opening directory “/etc/geoclue/conf.d/”: No such file or directory
Apr 07 09:20:02 nixos pipewire[1596]: mod.jackdbus-detect: Failed to receive jackdbus reply: org.freedesktop.DBus.Error.ServiceUnknown: The name org.jackaudio.service was not provided by any .service files
Apr 07 09:20:02 nixos org.gnome.Shell.desktop[1645]: Errors from xkbcomp are not fatal to the X server
Apr 07 09:20:04 nixos .gsd-sharing-wr[1482]: Failed to StopUnit service: GDBus.Error:org.freedesktop.DBus.Error.Spawn.ChildExited: Process org.freedesktop.systemd1 exited with status 1
Apr 07 09:20:04 nixos .gsd-sharing-wr[1482]: Failed to StopUnit service: GDBus.Error:org.freedesktop.DBus.Error.Spawn.ChildExited: Process org.freedesktop.systemd1 exited with status 1
Apr 07 09:20:06 nixos gnome-shell[2001]: libinput error: client bug: timer event6 hold: scheduled expiry is in the past (-45ms), your system is too slow
Apr 07 09:20:07 nixos .gnome-shell-wr[2001]: Error looking up permission: GDBus.Error:org.freedesktop.portal.Error.NotFound: No entry for geolocation
Apr 07 09:20:07 nixos .gsd-usb-protec[2199]: Failed to fetch USBGuard parameters: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.usbguard1 was not provided by any .service files
Apr 07 09:20:07 nixos gnome-shell[2510]: Errors from xkbcomp are not fatal to the X server
Apr 07 09:20:07 nixos gnome-shell[2521]: Errors from xkbcomp are not fatal to the X server
Apr 07 09:20:07 nixos pipewire[2443]: mod.jackdbus-detect: Failed to receive jackdbus reply: org.freedesktop.DBus.Error.ServiceUnknown: The name org.jackaudio.service was not provided by any .service files
Apr 07 09:20:08 nixos .gsd-housekeepi[1456]: Error releasing name org.gnome.SettingsDaemon.Housekeeping: The connection is closed
Apr 07 09:20:08 nixos .gsd-rfkill-wra[1442]: Error releasing name org.gnome.SettingsDaemon.Rfkill: The connection is closed
Apr 07 09:20:08 nixos .gnome-shell-wr[1316]: JS ERROR: Gio.IOErrorEnum: Xwayland exited unexpectedly
Apr 07 09:20:08 nixos .gsd-print-noti[1429]: Error releasing name org.gnome.SettingsDaemon.PrintNotifications: The connection is closed
Apr 07 09:20:10 nixos brave-browser.desktop[2599]: [2600:2600:0407/092010.202765:ERROR:policy_logger.cc(157)] :components/enterprise/browser/controller/chrome_browser_cloud_management_controller.cc(161) Cloud management controller initialization aborted as CBCM is not enabled. Please use the `--enable-chrome-browser-cloud-management` command line flag to enable it if you are not using the official Google Chrome build.
Apr 07 09:20:10 nixos brave-browser.desktop[2599]: [2600:2600:0407/092010.623523:ERROR:object_proxy.cc(576)] Failed to call method: org.freedesktop.ScreenSaver.GetActive: object_path= /org/freedesktop/ScreenSaver: org.freedesktop.DBus.Error.NotSupported: This method is not part of the idle inhibition specification: https://specifications.freedesktop.org/idle-inhibit-spec/latest/
Apr 07 09:20:16 nixos brave-browser.desktop[2599]: [2600:2600:0407/092016.913084:ERROR:interface_endpoint_client.cc(707)] Message 0 rejected by interface blink.mojom.WidgetHost
Apr 07 09:21:12 nixos brave-browser.desktop[2599]: [2637:2637:0407/092112.522195:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 1 times!
Apr 07 09:21:39 nixos brave-browser.desktop[2599]: [2637:2637:0407/092139.237203:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 2 times!
Apr 07 09:22:14 nixos brave-browser.desktop[2599]: [2637:2637:0407/092214.405093:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 3 times!
```

**Comment by aleeusgr** [comment]
thanks to propagandalf at nix discord.
```nix
{pkgs, ...}: {
  # Workaround for GNOME autologin: https://github.com/NixOS/nixpkgs/issues/103746#issuecomment-945091229
  systemd.services."getty@tty1".enable = false;
  systemd.services."autovt@tty1".enable = false;

  services.xserver.desktopManager.gnome.enable = true;
  services.udev.packages = with pkgs; [gnome.gnome-settings-daemon];
  environment.systemPackages = with pkgs; [gnomeExtensions.appindicator];
  environment.gnome.excludePackages =
    (with pkgs; [gnome-photos gnome-connections gnome-tour])
    ++ (with pkgs.gnome; [
      cheese # photo booth
      eog # image viewer
      epiphany # web browser
      gedit # text editor
      totem # video player
      yelp # help viewer
      file-roller # archive manager
      geary # email client
      gnome-calculator
      gnome-characters
      gnome-contacts
      gnome-maps
      gnome-music
      gnome-weather
    ]);
}
```
<@930457726323138570>

**Comment by aleeusgr** [comment]
try to set up proper driver:
https://nixos.org/manual/nixos/stable/#sec-x11-graphics-cards-nvidia

**Comment by aleeusgr** [comment]
try disabling Bluetooth as a workaround.

**Comment by aleeusgr** [comment]
`dmesg | grep -iE 'error|warn|fail'` | sort | uniq

**Comment by aleeusgr** [comment]
https://web.archive.org/web/20170904210829/http://scofaq.aplawrence.com:80/FAQ_scotec1haltcatch.html

**Comment by aleeusgr** [comment]
```shell
$ dmesg

...

[    3.461249] ACPI BIOS Error (bug): Could not resolve symbol [\CTDP], AE_NOT_FOUND (20230628/psargs-330)             
[    3.461729] ACPI Error: Aborting method \_SB.IETM.IDSP due to previous error (AE_NOT_FOUND) (20230628/psparse-529)  
[    3.462230] ACPI Warning: \_SB.IETM._TRT: Return Package has no elements (empty) (20230628/nsprepkg-94)             
[    3.464055] intel_pmc_core INT33A1:00:  initialized                                                                 
[    3.466282] Consider using thermal netlink events interface                                                         
[    3.484031] ACPI BIOS Error (bug): Could not resolve symbol [\_SB.PC00.LPCB.EC0.SEN1._CRT.S1CT], AE_NOT_FOUND (20230
628/psargs-330)                                                                                                        
[    3.484376] ACPI Error: Aborting method \_SB.PC00.LPCB.EC0.SEN1._CRT due to previous error (AE_NOT_FOUND) (20230628/
psparse-529)                                                                                                           
[    3.485063] ACPI BIOS Error (bug): Could not resolve symbol [\_SB.PC00.LPCB.EC0.SEN1._HOT.S1HT], AE_NOT_FOUND (20230
628/psargs-330) 
```

**Comment by aleeusgr** [comment]
```
$ journalctl -p 3 -x | cut -d ':' -f 2- | sort -u

...
54:29 nixos gdm-password][1855]: gkr-pam: unable to locate daemon control file                                         
54:31 nixos kernel:                                                                                                    
54:31 nixos kernel: ACPI BIOS Error (bug): Could not resolve symbol [\CTDP], AE_NOT_FOUND (20230628/psargs-330)

59:31 nixos pulseaudio[2555]: org.bluez.BatteryProviderManager1.UnregisterBatteryProvider() Failed: org.freedesktop.DBu
s.Error.NoReply:Message recipient disconnected from message bus without replying                                       
-- Boot 03736ad6f53949aa95e97b7b6e2e3941 --                                                                            
-- Boot 03b4225e5795494cb05800e1ad719811 --

```

**Comment by aleeusgr** [comment]
depends on #95
